radsat-sk/operation/subsystems/RTransceiver.c

```diff
@@ -0,0 +1,290 @@
+	uint16_t rxFrames = 0;
+	error = transceiverRxFrameCount(&rxFrames);
+	telemetry->rx.frames = rxFrames;
+
+	if (error)
+		return error;
+
+	// convert the raw receiver values to true telemetry values
+	// TODO: should this be done here? Or by Ground Station?
+	telemetry->rx.rx_doppler = rawRxTelemetry.fields.rx_doppler * (rawRxTelemetry.fields.rx_doppler * 0.00005887);
+	telemetry->rx.rx_rssi = rawRxTelemetry.fields.rx_rssi * (rawRxTelemetry.fields.rx_rssi * 0.00005887);
+	telemetry->rx.bus_volt = rawRxTelemetry.fields.bus_volt * 0.00488;
+	telemetry->rx.vutotal_curr = rawRxTelemetry.fields.vutotal_curr * 0.16643964;
+	telemetry->rx.vutx_curr = rawRxTelemetry.fields.vutx_curr * 0.16643964;
+	telemetry->rx.vurx_curr = rawRxTelemetry.fields.vurx_curr * 0.16643964;
+	telemetry->rx.vupa_curr = rawRxTelemetry.fields.vupa_curr * 0.16643964;
+	telemetry->rx.pa_temp = rawRxTelemetry.fields.pa_temp * -0.07669 + 195.6037;
+	telemetry->rx.board_temp = rawRxTelemetry.fields.board_temp * -0.07669 + 195.6037;
+
+	// convert the raw transmitter values to true telemetry values
+	// TODO: should this be done here? Or by Ground Station?
+	telemetry->tx.tx_reflpwr = rawTxTelemetry.fields.tx_reflpwr * 13.352 - 22300;
+    telemetry->tx.tx_fwrdpwr = rawTxTelemetry.fields.tx_fwrdpwr * 0.03 - 152;
+    telemetry->tx.bus_volt = rawTxTelemetry.fields.bus_volt * 0.00488;
+	telemetry->tx.vutotal_curr = rawTxTelemetry.fields.vutotal_curr * 0.16643964;
+	telemetry->tx.vutx_curr = rawTxTelemetry.fields.vutx_curr * 0.16643964;
+	telemetry->tx.vurx_curr = rawTxTelemetry.fields.vurx_curr * 0.16643964;
+	telemetry->tx.vupa_curr = rawTxTelemetry.fields.vupa_curr * 0.16643964;
+	telemetry->tx.pa_temp = rawTxTelemetry.fields.pa_temp * -0.07669 + 195.6037;
+	telemetry->tx.board_temp = rawTxTelemetry.fields.board_temp * -0.07669 + 195.6037;
+
+	uptime = 0;
+	error = transceiverTxUpTime(&uptime);
+	telemetry->tx.uptime = uptime;
+
+	return error;
+}
+
+
+/***************************************************************************************************
+                                         PRIVATE FUNCTIONS
+***************************************************************************************************/
+
+/**
+ * Gets the receiver's current up time.
+ *
+ * @param	uptime Receiver's up time in seconds.
+ * @return	Error code; 0 for success, otherwise see hal/errors.h.
+ */
+static int transceiverRxUpTime(uint16_t* uptime) {
+
+	int error = IsisTrxvu_rcGetUptime(TRANSCEIVER_INDEX, (unsigned int *) uptime);
+
+	// TODO: record errors (if present) to System Manager
+
+	return error;
+}
+
+
+/**
+ * Gets the transmitter's current up time.
+ * 
+ * @param	uptime transmitter's up time in seconds.
+ * @return	Error code; 0 for success, otherwise see hal/errors.h.
+ */
+static int transceiverTxUpTime(uint16_t* uptime) {
+
+	int error = IsisTrxvu_tcGetUptime(TRANSCEIVER_INDEX, (unsigned int *) uptime);
+
+	// TODO: record errors (if present) to System Manager
+
+	return error;
+}
+

```

Comment:
Need to set up beacon in this module with "set beacon" command

Reply:
See other comment; Beacon work is outside the scope of this PR since it's already getting pretty huge